framework/src/modules/nft/endpoint.ts

```diff
@@ -143,65 +147,83 @@ export class NFTEndpoint extends BaseEndpoint {
 		};
 	}
 
-	public async getCollectionIDs(
+	public async getSupportedCollectionIDs(
 		context: ModuleEndpointContext,
-	): Promise<{ collectionIDs: string[] }> {
-		const { params } = context;
+	): Promise<{ supportedCollectionIDs: string[] }> {
+		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
+		if (await supportedNFTsStore.has(context, ALL_SUPPORTED_NFTS_KEY)) {
+			return { supportedCollectionIDs: ['*'] };
+		}
 
-		validator.validate<{ chainID: string }>(getCollectionIDsRequestSchema, params);
+		const supportedCollectionIDs: string[] = [];
+		const supportedNFTsStoreData = await supportedNFTsStore.getAll(context);
 
-		const chainID = Buffer.from(params.chainID, 'hex');
+		for (const { key, value } of supportedNFTsStoreData) {
+			if (!value.supportedCollectionIDArray.length) {
+				supportedCollectionIDs.push(`${key.toString('hex')}********`);
+			} else {
+				const collectionIDs = value.supportedCollectionIDArray.map(
+					supportedCollectionID =>
+						key.toString('hex') + supportedCollectionID.collectionID.toString('hex'),
+				);
+				supportedCollectionIDs.push(...collectionIDs);
+			}
+		}
 
-		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
+		const nftStore = this.stores.get(NFTStore);
+		const { ownChainID } = this._config;
 
-		const chainExists = await supportedNFTsStore.has(context.getImmutableMethodContext(), chainID);
+		const allNFTs = await nftStore.iterate(context.getImmutableMethodContext(), {
+			gte: Buffer.alloc(LENGTH_NFT_ID, 0),
+			lte: Buffer.alloc(LENGTH_NFT_ID, 255),
+		});
 
-		if (!chainExists) {
-			return { collectionIDs: [] };
-		}
+		const nativeCollectionIDs = allNFTs
+			.filter(nft => nft.value.owner.equals(ownChainID))
+			.map(nft => nft.key.subarray(LENGTH_ADDRESS, LENGTH_NFT_ID).toString('hex'));
```

Comment:
LENGTH_ADDRESS is 20 and LENGTH_NFT_ID is 16. I am not sure what the intention was but probably subarray(20,16) wouldn't be meaningful ?

Reply:
Updated